src/users/components/UserList.component.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import UserCardComponent from './UserCard.component';
+import { InView } from 'react-intersection-observer';
+import { useQuery } from '@tanstack/react-query';
+import { getUserPagedUrl } from '../../config/urls.config';
+import { UserInterface } from '../interfaces/user.interface';
+import Loader from '../../common/loader.component';
+import AddUserCardComponent from './AddUserCardComponent';
+import { useNavigate } from 'react-router';
+import { CreateUserRoute } from '../../config/routes.config';
+import { PagedResponse } from '../../common/interfaces/pagedResponse.interface';
+
+export default function UserListComponent() {
+  const [page, setPage] = useState<number>(1);
+  const pagedData = useRef<PagedResponse<UserInterface>>({
+    data: [],
+    isNextAvaible: true,
+    ok: true
+  });
+  const navigate = useNavigate();
+
+  const { isLoading, error, data, refetch } = useQuery<unknown, unknown, PagedResponse<UserInterface>>(['user/pages', page], () =>
```

Comment:
Define the callback `() => fetch` in a separate file called `user-queries.ts` and make sure that's used here
A lot of inline stuff here